src/NuGet.Core/NuGet.Packaging/Signing/SignatureUtility.cs

```diff
@@ -0,0 +1,487 @@
+                        {
+                            throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateMultipleAttributes);
+                        }
+
+                        if (attribute.Values.Count != 1)
+                        {
+                            throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateMultipleAttributeValues);
+                        }
+
+                        signingCertificateAttribute = attribute;
+                        break;
+
+                    case Oids.SigningCertificateV2:
+                        if (signingCertificateV2Attribute != null)
+                        {
+                            throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateV2MultipleAttributes);
+                        }
+
+                        if (attribute.Values.Count != 1)
+                        {
+                            throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateV2MultipleAttributeValues);
+                        }
+
+                        signingCertificateV2Attribute = attribute;
+                        break;
+                }
+            }
+
+            var signatureType = AttributeUtility.GetCommitmentTypeIndication(signerInfo);
+            var isIssuerSerialRequired = false;
+
+            switch (signatureType)
+            {
+                case SignatureType.Author:
+                case SignatureType.Repository:
+                    if (signingCertificateAttribute != null)
+                    {
+                        throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateAttributeMustNotBePresent);
+                    }
+
+                    if (signingCertificateV2Attribute == null)
+                    {
+                        throw new SignatureException(errors.InvalidSignature, Strings.SigningCertificateV2AttributeMustBePresent);
+                    }
+
+                    isIssuerSerialRequired = true;
+                    break;
+            }
+
+            // This returns a new X509Certificate2Collection, which is mutable.
+            // Changes to this collection instance are not reflected back to SignedCms.Certificates.
+            var extraStore = signedCms.Certificates;
+
+            if (signingCertificateV2Attribute != null)
+            {
+                var reader = signingCertificateV2Attribute.ToDerSequenceReader();
+                var signingCertificateV2 = SigningCertificateV2.Read(reader);
+
+                return GetCertificates(
+                    signerInfo.Certificate,
+                    extraStore,
+                    signingCertificateV2,
+                    errors,
+                    includeChain,
+                    signingSpecifications,
+                    isIssuerSerialRequired);
+            }
+            else if (signingCertificateAttribute != null)
```

Comment:
Your Author and Repository signatures can't have both (since you prohibit the SigningCertificate(V1)).  But whatever other kinds could.  And you'll ignore this one.  I think the `else` could/should be removed, unless you want to elsewhere assert that they're not both present.